server/controllers/conversation.js

```diff
@@ -0,0 +1,97 @@
+const asyncHandler = require('express-async-handler');
+const Conversation = require('../models/Conversation');
+const Message = require('../models/Message');
+
+// @route POST /conversation/create
+// @desc create a new conversation
+// @access Private
+exports.createConversation = asyncHandler(async (req, res, next) => {
+  const senderProfileId = req.body.senderProfileId;
+  const recieverProfileId = req.body.recieverProfileId;
+  if (!senderProfileId || !recieverProfileId) {
+    res.status(400);
+    throw new Error('Missing sender profileId or reciever profileId !');
+  }
+  const existingConversation = await Conversation.findOne({
+    senderProfileId,
+    recieverProfileId,
+  });
+  if (existingConversation) {
+    res.status(400);
+    throw new Error('The conversation exists');
+  }
+
+  const conversation = await Conversation.create({
+    senderProfileId,
+    recieverProfileId,
+  });
+  //create the new Message with conversationId to store all text messages
+  const massage = await Message.create({ conversationId: conversation._id });
```

Comment:
It'd be good to confirm that the new message is successfully created in the database before returning success to the client. You can put a check for validity of `message`.

Reply:
Also, please make sure we are spelling properly. 

```suggestion
  const message = await Message.create({ conversationId: conversation._id });
```